core/src/main/java/org/elasticsearch/index/mapper/TokenCountFieldMapper.java

```diff
@@ -136,7 +136,7 @@ protected void parseCreateField(ParseContext context, List<IndexableField> field
 
         final int tokenCount;
         if (value == null) {
-            tokenCount = (Integer) fieldType().nullValue();
+            tokenCount = 0;
```

Comment:
This breaks the handling of `nullValue`. Instead we should check if `nullValue()`  is also null and in that case we can ignore the document completely (return without adding any field), otherwise we should count the number of tokens in the value that replace the null field.